pkg/util/mount/mount_linux.go

```diff
@@ -31,28 +32,80 @@ import (
 	"github.com/golang/glog"
 )
 
-const FlagBind = syscall.MS_BIND
-const FlagReadOnly = syscall.MS_RDONLY
+const (
+	// How many times to retry for a consistent read of /proc/mounts.
+	maxListTries = 3
+	// Number of fields per line in "/proc/mounts", as per the fstab man page.
+	expectedNumFieldsPerLine = 6
+)
 
 // Mounter implements mount.Interface for linux platform.
 type Mounter struct{}
 
-// Mount wraps syscall.Mount()
-func (mounter *Mounter) Mount(source string, target string, fstype string, flags uintptr, data string) error {
-	glog.V(5).Infof("Mounting %s %s %s %d %s", source, target, fstype, flags, data)
-	return syscall.Mount(source, target, fstype, flags, data)
+// Mount mounts source to target as fstype with given options. 'source' and 'fstype' must
+// be an emtpy string in case it's not required, e.g. for remount, or for auto filesystem
+// type, where kernel handles fs type for you. The mount 'options' currently come from
+// mount(8), if no more option is required, call Mount with an empty string list or nil.
+func (mounter *Mounter) Mount(source string, target string, fstype string, options []string) error {
+	glog.V(5).Infof("Mounting %s %s %s %v", source, target, fstype, options)
+	// Build mount command as follows:
+	//   mount [-t $fstype] [-o $options] [$source] $target
+	mountArgs := []string{}
+	if len(fstype) > 0 {
+		mountArgs = append(mountArgs, "-t", fstype)
+	}
+	if options != nil {
+		// If options is an empty list, this is a no-op.
+		mountArgs = append(mountArgs, options...)
```

Comment:
I think I'm with Tim (although not sure).  I think we should have an ops string[] which gets joined by a comma and a -o is put in front of it.

The only mount(8) argument I can think of that we might want is bind, and that can be a -o option instead
`mount --bind /a /b == mount -o bind /a /b`
Lets just not support the former...

mount -t nfs -o udp,ro,noatime,nodiratime server:/export /import 
    opts = ['udp','ro','noatime','nodiratime']

get done as:
```
mountArgs = append(mountArgs, "-o", strings.Join(args, ","))
```

Checking for valid options (if we so choose) becomes very easy.  (although I would suggest against)